src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -853,7 +853,6 @@ namespace AirLoopHVACDOAS {
         if (state.dataGlobal->BeginEnvrnFlag && this->MyEnvrnFlag) {
             bool ErrorsFound = false;
             Real64 rho;
-            state.dataSize->CurSysNum = this->m_OASystemNum;
```

Comment:
Underlying problem with the table reporting issues. This line reset CurSysNum to CurOASysNum which confused table reports. The component variables CurSysNum and CurOASysNum are set in SizingAirLoopDOAS as:

    state.dataSize->CurSysNum = state.dataHVACGlobal->NumPrimaryAirSys + this->m_AirLoopDOASNum + 1;
    state.dataSize->CurOASysNum = this->m_OASystemNum;

Whether these initializations should be moved to SimAirLoopHVACDOAS to emulate SimAirLoops is another question. I'm not sure these variables are used in anything other than sizing.